src/test/java/com/example/glovodb/service/OrderServiceTest.java

```diff
@@ -1,5 +1,159 @@
+import com.example.glovodb.repository.OrderRepository;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
 public class OrderServiceTest {
+    @Mock
+    public OrderRepository orderRepository;
+
+    @InjectMocks
+    private OrderService orderService;
+
+    private final int TEST_ID = 1;
+
+    @Test
+    public void createOrderTest() {
+        Order initialOrder = Order.builder().id(TEST_ID).name("Bob").build();
+        when(orderRepository.save(any())).thenReturn(initialOrder);
+        Order actualOrder = orderService.createOrder(new Order()).getBody();
+        assertEquals(initialOrder, actualOrder);
+    }
+
+    @Test
+    public void getOrderByIdTest() {
+        Order initialOrder = Order.builder().id(TEST_ID).name("Bob").build();
+        when(orderRepository.findById(any())).thenReturn(Optional.of(initialOrder));
+        Order actualOrder = orderService.getOrderById(anyInt()).getBody();
+        assertEquals(initialOrder, actualOrder);
+    }
+
+    @Test
+    public void getOrderByIdWhenIdNotFoundTest() {
+        when(orderRepository.findById(anyInt())).thenReturn(Optional.empty());
+        ResponseEntity<Order> response = orderService.getOrderById(TEST_ID);
+        assertEquals(HttpStatus.NOT_FOUND, response.getStatusCode());
+    }
+    @Test
+    public void updateOrderTest() {
+        Order initialOrder = Order.builder()
+                .id(TEST_ID)
+                .name("Bob")
+                .customerPhoneNumber("+123456")
+                .address("Olenivska str. 27")
+                .products(List.of(new Product()))
+                .build();
+        when(orderRepository.findById(any())).thenReturn(Optional.of(initialOrder));
+        Order actualOrder = orderService.updateOrder(anyInt(), initialOrder).getBody();
+        assertEquals(initialOrder, actualOrder);
+    }
+
+    @Test
+    public void updateOrderWhenOrderNotFoundTest() {
+        Order initialOrder = Order.builder().id(TEST_ID).build();
+        when(orderRepository.findById(anyInt())).thenReturn(Optional.empty());
+        ResponseEntity<Order> response = orderService.updateOrder(TEST_ID, initialOrder);
+        assertEquals(HttpStatus.NOT_FOUND, response.getStatusCode());
+    }
+
+    @Test
+    public void addProductToOrderTest() {
+        Collection<Product> products = new ArrayList<>();
+        products.add(new Product());
```

Comment:
You can simplify that by using `Collections.singletonList(new Product());` and past it to the products fields in builder

Reply:
I tried but got "java.lang.UnsupportedOperationException", because than I'm trying to add product to immutable collection.

Reply:
Yes, my bad, sorry.